src/backend/opencl/kernel/join.hpp

```diff
@@ -48,7 +48,7 @@ void join(Param out, const Param in, const af::dim4 offset) {
         std::ostringstream options;
         options << " -D To=" << dtype_traits<To>::getName()
                 << " -D Ti=" << dtype_traits<Ti>::getName()
-                << " -D dim=" << dim;
+                << " -D kDim=" << dim;
```

Comment:
`join.cl` OpenCL kernel doesn't use this one at all. I wonder why it's being passed to this at all.